pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go

```diff
@@ -456,6 +422,39 @@ func (gce *GCECloud) DeleteExternalTargetPoolAndChecks(name, region, clusterID s
 	return nil
 }
 
+// verifyUserRequestedIP checks the user-provided IP  to see whether it can be
+// used for the LB.  It also returns whether the IP is considered owned by the
+// user.
+func verifyUserRequestedIP(s CloudAddressService, region, requestedIP, fwdRuleIP, lbRef string) (ownedByUser bool, err error) {
+	if requestedIP == "" {
+		return false, nil
+	}
+	// If a specific IP address has been requested, we have to respect the
+	// user's request and use that IP. If the forwarding rule was already using
+	// a different IP, it will be harmlessly abandoned because it was only an
+	// ephemeral IP (or it was a different static IP owned by the user, in which
+	// case we shouldn't delete it anyway).
+	existingAddress, err := s.GetRegionAddressByIP(region, requestedIP)
+	if err != nil && !isNotFound(err) {
+		return false, fmt.Errorf("failed to check if the static IP %q exists: %v", requestedIP, err)
+	}
+	if err == nil {
+		// The requested IP is a static IP, owned and managed by the user.
+		glog.V(4).Infof("verifyUserRequestedIP: the requested static IP %q (name: %s) for LB %s exists.", requestedIP, existingAddress.Name, lbRef)
+		return true, nil
+	}
+	if requestedIP == fwdRuleIP {
+		// The requested IP is not a static IP, but is currently assigned
+		// to this forwarding rule, so we can just use it.
+		return false, nil
```

Comment:
Nope. See the original code [here](https://github.com/kubernetes/kubernetes/blob/58ea4e41d4947c0eff0b59685b29d86b3cbe50ae/pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go#L138-L147)